src/systems/filecoin_vm/sysactors/init_actor.go

```diff
@@ -133,45 +133,38 @@ func _codeIDSupportsExec(codeID actor.CodeID) bool {
 }
 
 func (a *InitActorCode_I) InvokeMethod(rt Runtime, method actor.MethodNum, params actor.MethodParams) InvocOutput {
-	argError := rt.ErrorReturn(exitcode.SystemError(exitcode.InvalidArguments))
+	_checkArgs := func(cond bool) {
+		if !cond {
+			rt.AbortArg()
+		}
+	}
 
 	switch method {
 	case actor.MethodConstructor:
-		if len(params) != 0 {
-			return argError
-		}
+		_checkArgs(len(params) == 0)
 		return a.Constructor(rt)
 
-	// case actor.MethodCron:
-	//     disable. init has no cron action
-
 	case Method_InitActor_Exec:
-		if len(params) == 0 {
-			return argError
-		}
+		_checkArgs(len(params) != 0)
+
 		codeId, err := actor.Deserialize_CodeID(Serialization(params[0]))
-		if err != nil {
-			return argError
-		}
+		_checkArgs(err == nil)
 		params = params[1:]
+
 		return a.Exec(rt, codeId, params)
 
 	case Method_InitActor_GetActorIDForAddress:
-		if len(params) == 0 {
-			return argError
-		}
+		_checkArgs(len(params) != 0)
+
 		address, err := addr.Deserialize_Address(Serialization(params[0]))
-		if err != nil {
-			return argError
-		}
+		_checkArgs(err == nil)
 		params = params[1:]
 
-		if len(params) != 0 {
-			return argError
-		}
+		_checkArgs(len(params) == 0)
```

Comment:
I don't get this construction - it's unnecessarily complicated. GetActorIDForAddress takes a single parameter. Just assert `len(params) == 1` and be done!

Reply:
The goal is that by following a standard pattern, even if it's slightly more verbose, the code can be less susceptible to errors when argument specs change over time. I've just updated the PR refactoring this logic out, though, so hopefully will be much cleaner.